tests/tests.rs

```diff
@@ -454,6 +454,61 @@ fn test_multiple_header_calls_in_builder() {
     }
 }
 
+#[test]
+fn test_multiple_header_contents() {
+    let actual = builder()
+        .header_contents("test.h", "int foo(const char* a);")
+        .header_contents("test2.h", "float foo2(const char* b);")
+        .clang_arg("--target=x86_64-unknown-linux")
+        .generate()
+        .unwrap()
+        .to_string();
+
+    let (actual, stderr) = rustfmt(actual);
+    println!("{}", stderr);
+
+    let (expected, _) = rustfmt(
+        "extern \"C\" {
+    pub fn foo2(b: *const ::std::os::raw::c_char) -> f32;
+}
+extern \"C\" {
+    pub fn foo(a: *const ::std::os::raw::c_char) -> ::std::os::raw::c_int;
+}
+"
+        .to_string(),
+    );
+
+    assert_eq!(expected, actual);
+}
+
+#[test]
+fn test_mixed_header_and_header_contents() {
+    let actual = builder()
+        .header(concat!(
+            env!("CARGO_MANIFEST_DIR"),
+            "/tests/headers/func_ptr.h"
+        ))
+        .header(concat!(env!("CARGO_MANIFEST_DIR"), "/tests/headers/char.h"))
+        .header_contents("test.h", "int bar(const char* a);")
+        .header_contents("test2.h", "float bar2(const char* b);")
+        .clang_arg("--target=x86_64-unknown-linux")
+        .generate()
+        .unwrap()
+        .to_string();
+
+    let (actual, stderr) = rustfmt(actual);
+    println!("{}", stderr);
+
+    let expected = include_str!(concat!(
+        env!("CARGO_MANIFEST_DIR"),
+        "/tests/expectations/tests/test_mixed_header_and_header_contents.rs"
```

Comment:
Hmm, how do you update this file when output changes? Do you need to do that manually? If so, it's a bit unfortunate.

Reply:
Hey, thanks for review :) And could you elaborate on this issue, please? I'm not sure what are the other ways to update the expectation but to do it manually, unfortunately. E.g., in test_multiple_header_calls_in_builder the pattern is the same, if I got it right.

Reply:
Yeah, you're right that this pattern is not new, it's just worse having tests you need to manually update, because the overhead of updating them scales with the amount of tests :( 

I guess we can take this as-is though.

Reply:
Actually, now that I think about it - I can use two existing test expectation files and try to concatenate them with inline expectation from header_contents calls, so that I will not introduce new expectation files.

Reply:
That sounds great. as long as the first header doesn't have any anonymous enum or struct it should work.